spec/services/preserved_object_handler_spec.rb

```diff
@@ -67,9 +67,78 @@
   end
 
   describe '#check_existence' do
-    # FIXME: move the actual specs to a separate file?
-    it 'stubbed spec for now' do
-      skip
+    context "(calls create or confirm_version)" do
+      it 'calls #create_after_validation when the object does not exit' do
+        expect(PreservedObject).to receive(:exists?).with(druid: druid).and_return(false)
+        no_exist_msg = "#{exp_msg_prefix} PreservedObject db object does not exist"
+        expect(Rails.logger).to receive(:log).with(Logger::ERROR, no_exist_msg)
+        # because create_after_validation isn't called, we don't check
+        # for the usual result codes it would return (doesn't exist and add to the db)
+        expect(po_handler).to receive(:create_after_validation)
+        po_handler.check_existence
+      end
+
+      it 'calls confirm_version when the object exists' do
+        expect(PreservedObject).to receive(:exists?).with(druid: druid).and_return(true)
+        expect(po_handler).to receive(:confirm_version)
+        po_handler.check_existence
+      end
+
+      it 'calls update_version_after_validation when confirm_version returns ARG_VERSION_GREATER_THAN_DB_OBJECT' do
```

Comment:
as stated above, I don't think we want to test that it calls the method, as much as we want to test that the outcome is what we want, with the correct db changes and the correct result_array. This sort of test is how we missed duplicate stuff.